crates/stages/src/stages/hashing_account.rs

```diff
@@ -155,96 +159,57 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
         // genesis accounts are not in changeset.
         if to_block - from_block > self.clean_threshold || from_block == 1 {
             let tx = provider.tx_ref();
-            let stage_checkpoint = input
-                .checkpoint
-                .and_then(|checkpoint| checkpoint.account_hashing_stage_checkpoint());
-
-            let start_address = match stage_checkpoint {
-                Some(AccountHashingCheckpoint { address: address @ Some(_), block_range: CheckpointBlockRange { from, to }, .. })
-                    // Checkpoint is only valid if the range of transitions didn't change.
-                    // An already hashed account may have been changed with the new range,
-                    // and therefore should be hashed again.
-                    if from == from_block && to == to_block =>
-                {
-                    debug!(target: "sync::stages::account_hashing::exec", checkpoint = ?stage_checkpoint, "Continuing inner account hashing checkpoint");
-
-                    address
-                }
-                _ => {
-                    // clear table, load all accounts and hash it
-                    tx.clear::<tables::HashedAccounts>()?;
 
-                    None
+            // clear table, load all accounts and hash it
+            tx.clear::<tables::HashedAccounts>()?;
+
+            let start = None;
+            let mut accounts_cursor = tx.cursor_read::<RawTable<tables::PlainAccountState>>()?;
+            let mut collector =
+                Collector::new(self.etl_config.file_size, self.etl_config.dir.clone());
+            let mut channels = Vec::with_capacity(10_000);
+
+            // itertools chunks doesn't support enumerate, so we use a counter to know when to flush
+            // hashes from channels to disk.
+            let mut flush_counter = 1;
+
+            // channels used to return result of account hashing
+            for chunk in &accounts_cursor.walk(start)?.chunks(100) {
```

Comment:
where are these numbers coming from?

Reply:
Made a few changes, but they're arbitrary. I tried this in some benchmarks, and didn't have apparent performance degradation.  We might want to tweak them in the future though.

Before we were just pushing all channels into vec, which would grow the memory usage immensily. So batching and flushing seems a more reasonable approach.